OSSM-8955 [DOC] Add text about using "latest" in the "version" field of the Istio CRD (#90860)
openshift/openshift-docs: modules/ossm-about-concepts-resources.adoc

----
//Helm charts can only be used with istio-csr and Helm charts are a temporary work around for istio-csr. This might get confusing for users, so a NOTE was added.

> To support canary updates of the control plane, {SMProduct} supports multiple {istio} versions. You can set the `version` field in the specification to the desired version, prefixed with `v`. To update, change this field to the new version. Use the `vX.Y-latest` alias to automatically select the latest patch version for a specific minor version. For example, setting `v1.23-latest` ensures the operator maintains the latest patch version of {istio} 1.23.

Reviewer: `Use the vX.Y-latest alias` sounds like it must be used. But it's optional. Probably better would be:
`You can set the full version or use a version alias vX.Y-latest which ....`

Author: @FilipB Made the change. PTAL. Thanks!

> To support canary updates of the control plane, {SMProduct} supports multiple {istio} versions. You can set the `version` field in the specification to the desired version, prefixed with `v`. To update, change this field to the new version. You can set the full version or use the `vX.Y-latest` alias to automatically select the latest patch version for a specific minor version. For example, setting `v1.23-latest` ensures the operator maintains the latest patch version of {istio} 1.23.

Reviewer 2: ```suggestion
To perform canary updates of the control plane, {SMProduct} supports multiple {istio} versions. You can set the `version` field to the new version by either using the full version or the `v<x>.<y>-latest` alias to automatically select the latest version for a specific minor version. For example, setting `v1.23-latest` ensures that the Operator maintains the latest version of {istio} 1.23.
```

- Suggested a few nits to minimalised the text.
-  _prefixed with `v`_ is redundant IMO because later it's suggested to use either the full version or the one with alias `v`
- _in the specification_ is not the exact resource-name/path-of the-field so IMO can be removed. You might want to provide "in the `xyz` resource or in the `spec.xyz.abc.version` field"
